internal/dag/gatewayapi_processor.go

```diff
@@ -1256,6 +1339,206 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
+						routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionType(cond.Type), cond.Status, gatewayapi_v1beta1.RouteConditionReason(cond.Reason), cond.Message)
+						continue
+					}
+					mirrorPolicy = &MirrorPolicy{
+						Cluster: &Cluster{
+							Upstream: mirrorService,
+						},
+					}
+				}
+			default:
+				routeAccessor.AddCondition(
+					gatewayapi_v1beta1.RouteConditionAccepted,
+					metav1.ConditionFalse,
+					gatewayapi_v1beta1.RouteReasonUnsupportedValue,
+					fmt.Sprintf("GRPCRoute.Spec.Rules.Filters: invalid type %q: only RequestHeaderModifier, ResponseHeaderModifier and RequestMirror are supported.", filter.Type),
+				)
+			}
+		}
+
+		// Priority is used to ensure if there are multiple matching route rules
+		// within an GRPCRoute, the one that comes first in the list has
+		// precedence. We treat lower values as higher priority so we use the
+		// index of the rule to ensure rules that come first have a higher
+		// priority. All dag.Routes generated from a single GRPCRoute rule have
+		// the same priority.
+		priority := uint8(ruleIndex)
+
+		// Get our list of routes based on whether it's a redirect or a cluster-backed route.
+		// Note that we can end up with multiple routes here since the match conditions are
+		// logically "OR"-ed, which we express as multiple routes, each with one of the
+		// conditions, all with the same action.
+		var routes []*Route
+
+		clusters, totalWeight, ok := p.GRPCClusters(route.Namespace, rule.BackendRefs, routeAccessor)
+		if ok {
+			routes = p.clusterRoutes(matchconditions, requestHeaderPolicy, responseHeaderPolicy, mirrorPolicy, clusters, totalWeight, priority, nil)
+		}
+
+		// Add each route to the relevant vhost(s)/svhosts(s).
+		for host := range hosts {
+			for _, route := range routes {
+				switch {
+				case listener.tlsSecret != nil:
+					svhost := p.dag.EnsureSecureVirtualHost(host)
+					svhost.Secret = listener.tlsSecret
+					svhost.AddRoute(route)
+				default:
+					vhost := p.dag.EnsureVirtualHost(host)
+					vhost.AddRoute(route)
+				}
+
+				programmed = true
+			}
+		}
+	}
+
+	return programmed, hosts
+}
+
+func gatewayGRPCMethodMatchCondition(match *gatewayapi_v1alpha2.GRPCMethodMatch, routeAccessor *status.RouteParentStatusUpdate) (MatchCondition, bool) {
+	if match == nil {
+		return nil, true
+	}
+
+	// Type specifies how to match against the service and/or method.
+	// Support: Core (Exact with service and method specified)
+	// Not Support: Implementation-specific (Exact with method specified but no service specified)
+	// Not Support: Implementation-specific (RegularExpression)
+
+	// Support "Exact" match type only. If match type is not specified, use "Exact" as default.
+	if match.Type != nil && *match.Type != gatewayapi_v1alpha2.GRPCMethodMatchExact {
+		routeAccessor.AddCondition(status.ConditionValidMatches, metav1.ConditionFalse, status.ReasonMethodMatchType, "GRPCRoute.Spec.Rules.Matches.Method: Only Exact match type is supported.")
```

Comment:
I think we can use an upstream condition type and reason here: `gatewayapi_v1beta1.RouteConditionAccepted` and `gatewayapi_v1beta1.RouteReasonUnsupportedValue`. It's not explicitly spelled out in the API spec (yet) for GRPCRoute, but this is what's typically used for other route types in these scenarios.